packages/backend/src/controllers/sqlRunnerController.ts

```diff
@@ -69,4 +73,72 @@ export class SqlRunnerController extends BaseController {
                 .getWarehouseFields(req.user!, projectUuid, tableName),
         };
     }
+
+    @Middlewares([
+        allowApiKeyAuthentication,
+        isAuthenticated,
+        unauthorisedInDemo,
+    ])
+    @SuccessResponse('200', 'Success')
+    @Post('/run')
+    @OperationId('runSql')
+    async runSql(
+        @Path() projectUuid: string,
+        @Body() body: SqlRunnerBody,
+        @Request() req: express.Request,
+    ): Promise<ApiJobScheduledResponse> {
+        this.setStatus(200);
+
+        return {
+            status: 'ok',
+            results: await this.services
+                .getProjectService()
+                .scheduleSqlJob(req.user!, projectUuid, body.sql),
+        };
+    }
+
+    /**
+     * Get results from a file stored locally
+     * @param fileId the fileId for the file
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Get('results/{fileId}')
+    @OperationId('getLocalResults')
+    async getLocalResults(
+        @Path() fileId: string,
+        @Path() projectUuid: string,
+        @Request() req: express.Request,
+    ): Promise<any> {
+        this.setStatus(200);
+        this.setHeader('Content-Type', 'application/json');
+
+        const resultsFilePath = await this.services
+            .getProjectService()
+            .getResultsFile(req.user!, projectUuid, fileId);
+
+        // TODO set response headers
```

Comment:
I've tried to add the right response headers, but they don't seem to work, I believe is a conflict with tsoa or sometihng. I can remove these comments and leave it for later.

Reply:
https://github.com/lukeautry/tsoa/issues/800